nemo/collections/common/prompts/formatter.py

```diff
@@ -20,22 +20,40 @@
 EOS_SLOT = "|eos|"
 
 
-class Modality(Enum):
+class BaseModalityType:
+    @staticmethod
+    def matches(value: Any) -> bool:
+        raise NotImplementedError
+
+
+class Text(BaseModalityType):
+    """Modality for text values."""
+
+    @staticmethod
+    def matches(value: str) -> bool:
+        return isinstance(value, str)
+
+
+class TextLiteral(BaseModalityType):
+    def __init__(self, *items):
+        self.allowed_values = items
+
+    def matches(self, value: str):
+        if "<|" in value and "|>" in value:
```

Comment:
`<|token|>` is a Canary-specific convention. This workaround shouldn't be inserted in a model-agnostic type such as this. IMO we don't need this workaround at all, simply add `<|pnc|>` and `<|nopnc|>` to Canary's `TextLiteral` slot types.